### ISSUE TITLE
Support references between requirements files

### ISSUE THREAD

**Issue by jab** [theoremlp/rules_uv#58]
requirements files [support referring to other requirements](https://pip.pypa.io/en/stable/reference/requirements-file-format/#example) (or constraints) files (by beginning a line with `-r` or `-c`). Composing requirements files thus is required to support correct, DRY, [layered requirements](https://pip-tools.readthedocs.io/en/stable/#workflow-for-layered-requirements) (e.g. test dependencies should be constrained by runtime dependencies).

I tried using a requirements.in file that uses `-c` with rules_uv's `pip_compile` rule, but didn't see any way to specify the other requirements files that it references e.g. as data dependencies, so unsurprisingly it failed when I ran the `pip_compile` target. Would it make sense to add support for this?

Thanks for maintaining rules_uv! <3

**Comment by mark-thm** [comment]
This ruleset aims to be a drop-in replacement for https://rules-python.readthedocs.io/en/stable/api/pip.html#compile-pip-requirements — is that what you’re presently using/could you share an example of your configuration?

**Comment by jab** [comment]
I know, rules_python's `compile_pip_requirements` doesn't support this yet either, though it should, since `pip-compile` supports this and documents layered requirements as a first-class supported workflow.

An example that should be fairly quick to reproduce would be to clone https://github.com/jab/shorty and make the following 1-line change:
```diff
diff --git a/requirements.test.in b/requirements.test.in
index b998a06..a5eb409 100644
--- a/requirements.test.in
+++ b/requirements.test.in
@@ -1 +1,2 @@
+-c ./requirements.base.txt
 absl-py
```
This allows the dependency resolution for the test environment to take into account the already-resolved dependencies of the base environment (so the test deps can be smoothly layered on top).

If you try to compile this with rules_uv, it currently fails:
```
❯ bazel run :compile_test_requirements
INFO: Analyzed target //:compile_test_requirements (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:compile_test_requirements up-to-date:
  bazel-bin/compile_test_requirements
INFO: Elapsed time: 0.087s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/compile_test_requirements
error: Error parsing included file in `requirements.test.in` at position 0
  Caused by: failed to read from file `./requirements.base.txt`
  Caused by: No such file or directory (os error 2)
```
Of course, if you directly invoke `uv` outside the sandbox to compile this, it works just fine, since it can access the referenced requirements file:
```
❯ uv pip compile --python bazel-bin/app.runfiles/.app.venv/bin/python requirements.test.in
Resolved 1 package in 2ms
# This file was autogenerated by uv via the following command:
#    uv pip compile --python bazel-bin/app.runfiles/.app.venv/bin/python requirements.test.in
absl-py==2.1.0
    # via -r requirements.test.in
```

**Comment by mark-thm** [comment]
Fixed in #63

**Comment by jab** [comment]
Thanks for fixing this! Looking forward to picking it up in the next release.

(Also, just corrected the first sentence in my previous comment - rules_python's `compile_pip_requirements` _does_ support this already.)
